crates/vm/levm/src/opcode_handlers/arithmetic.rs

```diff
@@ -227,16 +227,7 @@ impl VM {
         let base = current_call_frame.stack.pop()?;
         let exponent = current_call_frame.stack.pop()?;
 
-        let exponent_byte_size = (exponent.bits().checked_add(7).ok_or(VMError::Internal(
-            InternalError::ArithmeticOperationOverflow,
-        ))? as u64)
-            / 8;
-        let exponent_byte_size_cost = gas_cost::EXP_DYNAMIC_BASE
-            .checked_mul(exponent_byte_size.into())
-            .ok_or(VMError::GasCostOverflow)?;
-        let gas_cost = gas_cost::EXP_STATIC
-            .checked_add(exponent_byte_size_cost)
-            .ok_or(VMError::GasCostOverflow)?;
+        let gas_cost = gas_cost::exp_gas_cost(exponent).map_err(VMError::OutOfGasErr)?;
```

Comment:
You can do this here. If not, then keep it as it is.
```suggestion
        let gas_cost = gas_cost::exp_gas_cost(exponent)?;
```

Reply:
Oh ok, let's keep it then.